src/libcore/iter.rs

```diff
@@ -452,20 +452,17 @@ pub trait Iterator {
         Scan{iter: self, f: f, state: initial_state}
     }
 
-    /// Creates an iterator that maps each element to an iterator,
-    /// and yields the elements of the produced iterators.
+    /// Creates an iterator that maps an iterator returning function to
+    /// each element. Yields the elements of the produced iterators.
```

Comment:
The function (closure) maps each element to an iterator (actually to anything implementing IntoIterator).

"`.flat_map()` takes a function that maps each element into a new iterator, and yields all the elements of the produced iterators.”

Let's add something you said about denesting as well? Might as well explain more at least IMO.

“This can be useful do unravel a nested structure.”